### ISSUE TITLE
CPU slider capped at 64.

### ISSUE THREAD

**Issue by MR-Jarble** [FoldingAtHome/fah-web-client-bastet#119]
I am running multiple workstations that have far more cores than the system will allow. In v7 there was a cap of 32 cores but you could add more slots to fully engage the processor.
![Screenshot 2024-01-07 210912](https://github.com/FoldingAtHome/fah-web-client-bastet/assets/156045426/b7944055-2d7c-47eb-9a9f-15ed0e1adbe2)

**Comment by Hou5e** [comment]
Which version of FAH are you currently using?

In v8.1.x or v8.3.1, you can use 'Resource Groups' to partition CPU resources into something that looks like slots from v7.

It might be worth trying the latest FAH v8.3.1 Alpha version until it gets released to Beta: https://foldingathome.org/alpha
For making Resource Groups in v8.3.1, it is similar but looks a bit different to these instructions: https://foldingforum.org/viewtopic.php?t=39101

Also, if you do try the alpha version, make sure to do a 'refresh without cache' in your web browser (typically: CTRL+SHIFT+R, or CRTL+F5) after you update.

**Comment by MR-Jarble** [comment]
Thanks for the reply. I was able to get resource groups to function as a work around but it would still be preferable to have the system correctly detect the number of cores.

**Comment by kbernhagen** [comment]
Please post the output of
```
fah-client --info
```

**Comment by MR-Jarble** [comment]
![Screenshot 2024-01-13 024758](https://github.com/FoldingAtHome/fah-web-client-bastet/assets/156045426/52ab4f52-211d-4154-bcfe-30a508ec009a)

**Comment by kbernhagen** [comment]
So it’s 64c/128t ?

**Comment by MR-Jarble** [comment]
Each cpu is 64 cores 128 threads for 128 cores and 256 threads

**Comment by kbernhagen** [comment]
If you or someone you know is a developer, maybe you could glance at the code?

Line 98
SystemInfo::getCPUCount()
https://github.com/CauldronDevelopmentLLC/cbang/blob/master/src/cbang/os/SystemInfo.cpp

**Comment by jcoffland** [comment]
It sounds like the issue is that you've got two CPUs and we're only counting the threads on one of them.

**Comment by jcoffland** [comment]
Does TaskManager show 128 cores?

**Comment by jcoffland** [comment]
I took a closer look that the Windows code for counting CPU cores and made some changes that could solve the problem.  @MR-Jarble Can you try out this build and tell me if it works any better?

    https://master.foldingathome.org/builds/fah-client/windows-10-64bit/release/v8.3.2-b64/fah-client_8.3.2_AMD64.exe

**Comment by MR-Jarble** [comment]
![image](https://github.com/FoldingAtHome/fah-web-client-bastet/assets/156045426/6fb2bd03-7bb7-4b86-9ca2-52cc8398e44d)
Still holding at 64 cores. I did a ctrl f5 to clear cache but let me know if I need to take any other steps.

**Comment by jcoffland** [comment]
Does this build work?  https://master.foldingathome.org/builds/fah-client/windows-10-64bit/release/v8.3.2-b65/fah-client_8.3.2_AMD64.exe

**Comment by jcoffland** [comment]
Does this build work?  https://master.foldingathome.org/builds/fah-client/windows-10-64bit/release/v8.3.2-b66/fah-client_8.3.2_AMD64.exe

**Comment by MR-Jarble** [comment]
> Does this build work? https://master.foldingathome.org/builds/fah-client/windows-10-64bit/release/v8.3.2-b65/fah-client_8.3.2_AMD64.exe

Negative it did not work. Exact same behavior as before with a cap at 64 cores.

**Comment by jcoffland** [comment]
v8.3.2-b65 was broken and would have failed to run at all.  Did you install v8.3.2-b66 or b65 as quoted above?

Sorry I posted the broken one, deleted it then posted the correct one.  You would have gotten to very similar looking emails.

**Comment by MR-Jarble** [comment]
Ahh I see now let me try again

**Comment by MR-Jarble** [comment]
Yep that was my bad I did not see the b65 vs b66 path change. New one now shows a full count of 256 cores! 
![image](https://github.com/FoldingAtHome/fah-web-client-bastet/assets/156045426/b48b98a3-00e3-417c-9de8-e4cff7c71412)

**Comment by jcoffland** [comment]
Is that right?  We want to count CPU cores and not threads.  It's not helpful to run F@H with one thread for each CPU thread.  We want to run one thread for each CPU core.  In your case it should be 128, right?

**Comment by jcoffland** [comment]
What do you get from this build?  https://master.foldingathome.org/builds/fah-client/windows-10-64bit/release/v8.3.2-b72/fah-client_8.3.2_AMD64.exe

**Comment by kbernhagen** [comment]
> Is that right? We want to count CPU cores and not threads. It's not helpful to run F@H with one thread for each CPU thread. We want to run one thread for each CPU core. In your case it should be 128, right?

It should be available logical cpus, which DOES count threads.

**Comment by kbernhagen** [comment]
Historically it has been threads. 
People with 8c/16t will see benefit running more than CPUs 8.

**Comment by kbernhagen** [comment]
It has been up to user to limit CPUs available to fah.

**Comment by kbernhagen** [comment]
It might be reasonable on big.little to only count p-cores. But that would disappoint people who actually want folding running on e-cores. 

It would also break the client if someone makes a cpu with only e-cores.

**Comment by kbernhagen** [comment]
A separate real core count method could be useful if we want to present user with a preferred cpu count.

**Comment by jcoffland** [comment]
You're right it should be threads not cores.

**Comment by MR-Jarble** [comment]
Just to jump back in I agree that it should be threads not cores so that the user can balance the load. For instance my quad xeon server is able to take three 64 (seems to be the current max work load size) cpu work loads and balance them for best performance. The old machine is putting out work units around the one hour mark so not bad at all.
